test/unit/JetStakingV3.test.ts

```diff
@@ -0,0 +1,2274 @@
+            await auroraToken.connect(auroraOwner).transfer(user2.address, ethers.utils.parseUnits("10000", 18))
+            await auroraToken.connect(auroraOwner).transfer(user3.address, ethers.utils.parseUnits("10000", 18))
+            await auroraToken.connect(auroraOwner).transfer(user4.address, ethers.utils.parseUnits("10000", 18))
+            await auroraToken.connect(auroraOwner).transfer(stakingAdmin.address, ethers.utils.parseUnits("100000000", 18))
+            await auroraToken.connect(auroraOwner).transfer(streamManager.address, ethers.utils.parseUnits("100000000", 18))
+            // console.log(balanceOfAurorOwner)
+            // transfer 20% of the total supply to the treasury contract
+            const twentyPercentOfAuroraTotalSupply = ethers.utils.parseUnits("200000000", 18)
+            // const onePercentOfTokenSupply = ethers.utils.parseUnits("1000000", 18)
+            await auroraToken.connect(auroraOwner).transfer(treasury.address, twentyPercentOfAuroraTotalSupply)
+            const balanceOfAurorOwner = await auroraToken.balanceOf(auroraOwner.address)
+            await auroraToken.connect(auroraOwner).transfer(user5.address, balanceOfAurorOwner.sub(ethers.utils.parseUnits("1000000", 18)))
+            // transfer ownership of the treasury to the jet staking contract
+            await treasury.connect(auroraOwner).grantRole(defaultAdminRole, jet.address)
+        }
+    );
+
+    async function reset(snapshotId: string) {
+        await network.provider.request({
+          method: 'evm_revert',
+          params: [snapshotId] // snapshot is global
+        });
+    }
+
+    it('should test multiple stakers reward calculation', async () => {
+        await setupTest();
+        const id = 1
+        // approve aurora tokens to the stream proposal
+        const auroraProposalAmountForAStream = 0
+        const maxRewardProposalAmountForAStream = ethers.utils.parseUnits("200000000", 18)
+        const minRewardProposalAmountForAStream = ethers.utils.parseUnits("100000000", 18)
+        await auroraToken.connect(streamManager).approve(jetV3.address, auroraProposalAmountForAStream)
+        const amount1 = ethers.utils.parseUnits("1", 0)
+        const amount2 = ethers.utils.parseUnits("1002", 18)
+        const totalDeposit = amount1.add(amount2)
+        await auroraToken.connect(user1).approve(jetV3.address, amount1)
+        await auroraToken.connect(user2).approve(jetV3.address, amount2)
+        await network.provider.send("evm_setAutomine", [false])
+        // Users stake in the same block to compare rewards with the same stream weight
+        await jetV3.connect(user1).stake(amount1)
+        await jetV3.connect(user2).stake(amount2)
+        await network.provider.send("evm_mine")
+        await network.provider.send("evm_setAutomine", [true])
+        // create a stream
+        const startTime = (await ethers.provider.getBlock("latest")).timestamp + 100
+        const scheduleTimes = [
```

Comment:
This tests only for 5 `scheduledRewards`, can we please extend it to find max numbers of `Schedules` we can accommodate in a `Stream`

Reply:
The test you mean is here https://github.com/aurora-is-near/aurora-staking-contracts/pull/178/files#diff-1b80ceba3723623352fae46343a27efe7f02daf02e9c11e20ebae36dab1d1b7aR2229 for 200 rewards items.

Reply:
This means max 200 values can be stored in `scheduledRewards`? I think, the number could be greater than that, to check, can you please include a failing scenario for `one + max size found`?

Reply:
using `one + max size` is gonna overflow! + it is computationally infeasible for a solidity contract. We can increase the number but I don't think that giving the current requirements for a schedule won't exceed 200 quarters (~50 years).